### DISCUSSION TITLE
Keep config.toml in sync between local <> prod

### DISCUSSION THREAD

**Discussion by sp88011** [supabase/supabase#22455] [Questions]
How would I go about pulling the config.toml that underlies my production database so that I can apply it (in some way) to my local environment?

Or put more generally, how am I supposed to work with the production and local config.toml files?

Example:
-  I create a new email template using the Supabase dashboard in the production instance. How do I get those changes reflected in my local config.toml?

**Comment by vipulrawat** [comment]
I feel bad that there is no mention about this in documentation (atleast I couldn't find it easily).

**Comment by vinicentus** [comment]
When running `supabase link`, and the remote project has been edited before setting up the local project, I get this warning:
`WARNING: Local config differs from linked project. Try updating supabase/config.toml`.

I can't find any info on if it needs resolving, the importance of it, or how to resolve it.

<details>
  <summary>Full log of the command</summary>

```
[vincent@computer supabase-project]$ supabase link --project-ref redacted
Enter your database password (or leave blank to skip): 
Connecting to remote database...
Finished supabase link.
WARNING: Local config differs from linked project. Try updating supabase/config.toml
[api]
enabled = true
port = 54321
schemas = ["public", "graphql_public"]
extra_search_path = ["public", "extensions"]
max_rows = 1000
[api.tls]
enabled = false
```
</details>

**Reply by Michota** [reply]
> When running `supabase link`, and the remote project has been edited before setting up the local project, I get this warning: `WARNING: Local config differs from linked project. Try updating supabase/config.toml`.
> 
> I can't find any info on if it needs resolving, the importance of it, or how to resolve it.
> 
> Full log of the command

Same, but what's funny: i got no config.toml, so it shouldn't really complain and it should create `config.toml` from scratch based on what's on remote

**Comment by JorSoi** [comment]
any news on that ?

**Comment by efitzkiwi** [comment]
Sadly the best option I have found was to roll my own config manager, and read the local config.toml into a middleware (like a go application) and parse the values, then sync via supabase CLI to your upstream DBs.

**Reply by alexryanterry** [reply]
@efitzkiwi - is this something you can share so we can use (and improve?) as well?

**Comment by robwhess** [comment]
I made a feature request for this feature in #34456.
